service/src/main/java/org/databiosphere/workspacedataservice/service/TsvStreamWriteHandler.java

```diff
@@ -57,9 +57,13 @@ public TsvStreamWriteHandler(InputStream inputStream, ObjectReader tsvReader, Re
 			colNames = StreamSupport.stream(actualSchema.spliterator(), false)
 					.map(CsvSchema.Column::getName)
 					.toList();
-
-			if (colNames.stream().anyMatch(col -> Collections.frequency(colNames, col) > 1)) {
-				throw new InvalidTsvException("TSV contains duplicate column names."
+			// any blank headers or trailing tabs is considered malformed 
+			if(colNames.stream().anyMatch(col -> col.isBlank())) {
```

Comment:
nit: to avoid warnings in IntelliJ about "Lambda can be replaced with method reference":
```suggestion
			if(colNames.stream().anyMatch(String::isBlank)) {
```

Reply:
Oh cool I didn't realize you could do that.